scr/src/main/java/org/apache/felix/scr/impl/logger/ScrLogManager.java

```diff
@@ -0,0 +1,293 @@
+			return new ExtLogManager(context, config).scr();
+		else
+			return new ScrLogManager(context, config).scr();
+	}
+
+	ScrLogManager(BundleContext context, ScrConfiguration config) {
+		super(context);
+		this.config = config;
+		this.bundle = context.getBundle();
+	}
+
+	/**
+	 * This logger is used for the main code of SCR. This will use the SCR
+	 * bundle & the {@link Logger#ROOT_LOGGER_NAME}
+	 * 
+	 * @return an Scr Logger.
+	 */
+	public ScrLogger scr() {
+		ScrLoggerFacade scrl = super.getLogger(bundle, Logger.ROOT_LOGGER_NAME, ScrLoggerFacade.class);
+		scrl.setPrefix(getBundleIdentifier(bundle));
+		return scrl;
+	}
+
+	/**
+	 * This logger is used for the logging on a per bundle basis. This will use
+	 * the target bundle & the {@link Logger#ROOT_LOGGER_NAME}
+	 * 
+	 * @param bundle
+	 *                   the target bundle
+	 * @return a logger suitable to log bundle entries
+	 */
+	public BundleLogger bundle(Bundle bundle) {
+		ScrLoggerFacade logger = getLogger(bundle, Logger.ROOT_LOGGER_NAME, ScrLoggerFacade.class);
+		logger.setPrefix(getBundleIdentifier(bundle));
+		return logger;
+	}
+
+	/**
+	 * This logger is used for the logging on a per bundle basis. This will use
+	 * the target bundle & the implementation class as logger name.
+	 * 
+	 * @param bundle
+	 *                   the target bundle
+	 * @return a logger suitable to log bundle entries
+	 */
+	public ComponentLogger component(Bundle bundle, String implementationClass, String name) {
+
+		// assert bundle != null;
+		// assert bundle.getSymbolicName() != null : "scr requires recent bundles";
+		// assert implementationClass != null;
+		// assert name != null;
+
+		ScrLoggerFacade facade = getLogger(bundle, implementationClass, ScrLoggerFacade.class);
+		facade.setComponentId(-1);
+		return (ComponentLogger) facade;
+	}
+
+	class ScrLoggerFacade extends LoggerFacade implements InternalLogger, ScrLogger, BundleLogger, ComponentLogger {
+		ScrLoggerFacade(LogDomain logDomain, String name) {
+			super(logDomain, name);
+		}
+
+		@Override
+		public void setComponentId(long id) {
+			setPrefix(componentPrefix(this, id));
+		}
+
+		public boolean isLogEnabled(Level level) {
+
+			// assert !closed.get();
+
+			Logger logger = getLogger();
+			if (logger != null) {
+				switch (level) {
+				case AUDIT:
+					return true;
+				case ERROR:
+					return logger.isErrorEnabled();
+				case WARN:
+					return logger.isWarnEnabled();
+				case INFO:
+					return logger.isInfoEnabled();
+				case TRACE:
+					return logger.isTraceEnabled();
+				case DEBUG:
+				default:
+					return logger.isDebugEnabled();
+				}
+			} else {
+				return getLogLevel().implies(level);
+			}
+		}
+
+		@Override
+		public void log(Level level, String format, Throwable ex, Object... arguments) {
+			log(level, format(format, arguments), ex);
+		}
+
+		@Override
+		public void log(Level level, String message, Throwable ex) {
+			if (prefix != null && prefix.length() > 0) {
```

Comment:
Can move this into the if check for `isLogEnabled(level)` to avoid unnecessary work.